File: 206-reverse-linked-list/reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    let lastNode = null;
    let currentNode = head;
    while (currentNode !== null){
        let next = currentNode.next;
        currentNode.next = lastNode;
        lastNode = currentNode;
        currentNode = next;
    }
    return lastNode;
};

File: 206-reverse-linked-list/reverse-linked-list.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @return {ListNode}
def reverse_list(head)
    last_node = nil
    current_node = head
    next_node = head&.next
    while current_node != nil
        current_node.next = last_node
        last_node = current_node
        current_node = next_node
        next_node = next_node&.next
    end
    return last_node
end